packages/data-provider/src/services/validationService.ts

```diff
@@ -0,0 +1,45 @@
+import { BaseDependencies } from '../config/config.js';
+import { NotFound } from '../utils/errors.js';
+import { and, count, eq ,sql} from 'drizzle-orm/sql';
+import { submittedData } from '@overture-stack/lyric-data-model/models';
+const validationService = (dependencies: BaseDependencies) => {
+    const { db, logger } = dependencies; 
+    const LOG_MODULE = 'VALIDATION_SERVICE';
+
+    return {
+        validateRecord: async ({
+            categoryId,
+            entityName,
+            studyId,
+            value,
+        }: {
+            categoryId: number;
+            entityName: string;
+            studyId: string;
+            value: string;
+        }): Promise<boolean> => {
+            logger.debug(LOG_MODULE, 'Validating record', { categoryId, entityName, studyId, value });
+
+            try {
+
+                const resultCount = await db
+                    .select({ total: count() })
+                    .from(submittedData) 
+                    .where(
+                        and(
+                            eq(submittedData.dictionaryCategoryId, categoryId),
+                            eq(submittedData.entityName, entityName), 
+                            eq(submittedData.organization, studyId),
+                            sql`${submittedData.data} @> ${JSON.stringify({ [entityName]: value })}::jsonb`
```

Comment:
In this query, `[entityName]` should come from a configurable variable, this variable should be a string type that matches a field name within the entity of the category.

For instance, 
the categoryId `1`, has entity `participant`,  within participant properties there is one called `participant_id` field.

Via Configuration we set category: `1` entity: `participant` and field: `participant_id`

So if Request is category `1`. and entity `participant`, we will know based on configuration given to use field `participant_id` in this query. The resulting query will be:
`WHERE categoryId=1 AND entityName='particpant' AND data @> '{"particpant_id": "123"}'`